snakemake_storage_plugin_xrootd/__init__.py

```diff
@@ -16,178 +21,170 @@
 from snakemake_interface_storage_plugins.storage_object import (
     StorageObjectRead,
     StorageObjectWrite,
-    StorageObjectGlob,
     retry_decorator,
 )
 from snakemake_interface_storage_plugins.io import IOCacheStorageInterface
 
 
-def parse_query_params(query_params: List[str]):
-    """Parse query params from command line."""
-    return dict(param.split("=") for param in query_params)
+class XRootDFatalException(Exception):
+    """
+    Used to prevent retries for certain XRootD errors with the retry decorator
+    """
+
+    pass
 
 
-def unparse_query_params(query_params: Mapping[str, str]):
-    """Unparse query params to command line."""
-    return [f"{key}={value}" for key, value in query_params.items()]
+def _raise_fatal_error(exception: Type[Exception]):
+    if isinstance(exception, XRootDFatalException):
+        get_logger().warning("Unrecoverable error, no more retries")
+        raise exception
+
+
+# TODO configurable would be nice
+xrootd_retry = retry(
+    tries=3, delay=3, backoff=2, logger=get_logger(), fail_callback=_raise_fatal_error
+)
 
 
 @dataclass
 class StorageProviderSettings(StorageProviderSettingsBase):
-    host: Optional[str] = field(
-        default=None,
+    keep_local: Optional[bool] = field(
+        default=False,
         metadata={
-            "help": "The XrootD host to connect to",
+            "help": "Keep local copy of storage object(s)",
             "env_var": False,
-            "required": True,
-        },
-    )
-    port: Optional[int] = field(
-        default=None,
-        metadata={
-            "help": "The XrootD port to connect to",
-            "env_var": False,
-            "required": False,
-        },
-    )
-    username: Optional[str] = field(
-        default=None,
-        metadata={
-            "help": "The username to use for authentication",
-            "env_var": True,
             "required": False,
+            "type": bool,
         },
```

Comment:
This should not be a setting per plugin. Instead, this is handled globally by Snakemake itself.

Reply:
Okay makes sense! I'll remove it